Applications/spire/source/charting/cached_chart_model.cpp

```diff
@@ -0,0 +1,85 @@
+#include "spire/charting/cached_chart_model.hpp"
+
+using namespace boost::signals2;
+using namespace Spire;
+
+CachedChartModel::CachedChartModel(ChartModel& model)
+    : m_chart_model(&model) {}
+
+ChartValue::Type CachedChartModel::get_x_axis_type() const {
+  return m_chart_model->get_x_axis_type();
+}
+
+ChartValue::Type CachedChartModel::get_y_axis_type() const {
+  return m_chart_model->get_y_axis_type();
+}
+
+QtPromise<std::vector<Candlestick>> CachedChartModel::load(ChartValue first,
+    ChartValue last) {
+  auto range = ChartRange({first, last});
+  auto range_iterator = std::lower_bound(m_ranges.begin(), m_ranges.end(),
+    range, [=] (const auto& value, const auto& search_value) {
+      return value.m_start >= search_value.m_start &&
```

Comment:
Why is the lambda capturing by value?

This is also not a valid comparator. A comparator must define a strict weak ordering relation, in particular it must be the case that ```forall a: comp(a, a) == false```, however the comparator you provide has the property that ```forall a: comp(a, a) == true```:

https://en.cppreference.com/w/cpp/named_req/Compare

Reply:
What difference is capturing by value or reference here when it's not using any variables from the outer scope in the body?

Reply:
If you're not capturing anything then omit the capture entirely, ie, [] (const auto& lhs, const auto& rhs) { ... }.